// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";
import "pds_option.proto";
import "builtin_format.proto";
package baidu.paddle_serving.predictor.ctr_prediction;

option cc_generic_services = true;

message CTRReqInstance {
  repeated int64 sparse_ids = 1;
  repeated float dense_ids = 2;
};

message Request { repeated CTRReqInstance instances = 1; };

message CTRResInstance {
  required float prob0 = 1;
  required float prob1 = 2;
};

message Response {
  repeated CTRResInstance predictions = 1;
  required int64 err_code = 2;
  optional string err_msg = 3;
};

service CTRPredictionService {
  rpc inference(Request) returns (Response);
  rpc debug(Request) returns (Response);
  option (pds.options).generate_stub = true;
};
